spec/graphql/resolvers/customers/usage_resolver_spec.rb

```diff
@@ -101,12 +102,180 @@
+      expect(charge_usage['billableMetric']['code']).to eq(metric.code)
       expect(charge_usage['billableMetric']['aggregationType']).to eq('count_agg')
       expect(charge_usage['charge']['chargeModel']).to eq('graduated')
       expect(charge_usage['units']).to eq(4.0)
       expect(charge_usage['amountCents']).to eq('5')
     end
   end
+
+  context 'with one dimension group' do
+    let(:aws) { create(:group, billable_metric: metric, key: 'cloud', value: 'aws') }
+    let(:google) { create(:group, billable_metric: metric, key: 'cloud', value: 'google') }
+    let(:charge) do
+      create(
+        :standard_charge,
+        plan: subscription.plan,
+        billable_metric: metric,
+        properties: {},
+        group_properties: [
+          build(
+            :group_property,
+            group: aws,
+            values: { amount: '10', amount_currency: 'EUR' },
+          ),
+          build(
+            :group_property,
+            group: google,
+            values: { amount: '20', amount_currency: 'EUR' },
+          ),
+        ],
+      )
+    end
+
+    before do
+      create_list(
+        :event,
+        3,
+        organization: organization,
+        customer: customer,
+        subscription: subscription,
+        code: metric.code,
+        timestamp: Time.zone.now,
+        properties: { cloud: 'aws' },
+      )
+
+      create(
+        :event,
+        organization: organization,
+        customer: customer,
+        subscription: subscription,
+        code: metric.code,
+        timestamp: Time.zone.now,
+        properties: { cloud: 'google' },
+      )
+    end
+
+    it 'returns the group usage for the customer' do
+      result = execute_graphql(
+        current_user: membership.user,
+        current_organization: organization,
+        query: query,
+        variables: {
+          customerId: customer.id,
+          subscriptionId: subscription.id,
+        },
+      )
+
+      charge_usage = result['data']['customerUsage']['chargesUsage'][0]
+      groups_usage = charge_usage['groups']
+
+      aggregate_failures do
+        expect(charge_usage['units']).to eq(4)
+        expect(charge_usage['amountCents']).to eq('5000')
+        expect(groups_usage).to match_array(
+          [
+            { 'id' => aws.id, 'key' => nil, 'value' => 'aws', 'units' => 3, 'amountCents' => '3000' },
+            { 'id' => google.id, 'key' => nil, 'value' => 'google', 'units' => 1, 'amountCents' => '2000' },
+          ],
+        )
+      end
+    end
+  end
+
+  context 'with two dimensions group' do
+    let(:aws) { create(:group, billable_metric: metric, key: 'cloud', value: 'aws') }
+    let(:google) { create(:group, billable_metric: metric, key: 'cloud', value: 'google') }
+    let(:aws_usa) { create(:group, billable_metric: metric, key: 'region', value: 'usa', parent_group_id: aws.id) }
+    let(:aws_france) { create(:group, billable_metric: metric, key: 'region', value: 'france', parent_group_id: aws.id) }
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)